mern-ecommerce/controllers/order.js

```diff
@@ -61,3 +61,20 @@ exports.updateOrderStatus = (req, res) => {
     }
   );
 };
+
+exports.removeOrder = (req, res) => { 
+  Order.findByIdAndDelete(req.params.orderId) 
+    .then(deletedOrder => { 
+      if (!deletedOrder) {
```

Comment:
The condition `if (!deletedOrder)` is not checking if the order was successfully deleted, but rather if the order was found before deletion. It would be more accurate to check if the deletion was successful, for example, by checking the result of `findByIdAndDelete`. Additionally, consider adding a log or a more informative error message when the order is not found.